File: FujiIS/FujiIS/Class/clsWinApp.vb
Imports System
Imports System.IO
Imports System.Globalization

<System.Diagnostics.DebuggerStepThrough()>
Public Class AppContext
	Inherits ApplicationContext

	Private Shared instance_ As AppContext = New AppContext

	Private Sub New()
		MyBase.New()
	End Sub

	Public Shared Function GetInstance() As AppContext
		Return instance_
	End Function

	Public Sub SetNextContext(ByVal mainForm As Form, Optional ByVal toHide As Boolean = False)

		Dim tempForm As Form = Me.MainForm
		Dim orgCur As Cursor = Cursor.Current
		Cursor.Current = Cursors.WaitCursor

		Me.MainForm = mainForm

		If toHide Then
			Try
				If Not tempForm Is Nothing Then
					tempForm.Hide()
				End If
			Catch ex As Exception

				Call OutputLogFile("発生場所：" & Reflection.MethodBase.GetCurrentMethod.Name & vbNewLine & ex.Message)
				MessageBox.Show("発生場所：" & Reflection.MethodBase.GetCurrentMethod.Name & vbNewLine & ex.Message, "エラー", MessageBoxButtons.OK, MessageBoxIcon.Error)

			End Try
		Else
			Try

				tempForm.Close()

			Catch ex As Exception

				Call OutputLogFile("発生場所：" & Reflection.MethodBase.GetCurrentMethod.Name & vbNewLine & ex.Message)
				MessageBox.Show("発生場所：" & Reflection.MethodBase.GetCurrentMethod.Name & vbNewLine & ex.Message, "エラー", MessageBoxButtons.OK, MessageBoxIcon.Error)

			End Try
		End If

		Me.MainForm.Show()
		tempForm = Nothing
		Cursor.Current = orgCur

	End Sub
End Class

''' <summary>
''' メインルーチン
''' </summary>
Public Class clsWinApp

	Private Shared MyMutex As System.Threading.Mutex

	<STAThread()> Public Shared Sub Main()
		'ビジュアルスタイルを有効にする
		Application.EnableVisualStyles()

		Initialize()

	End Sub

	''' <summary>
	''' 例外ハンドラ
	''' </summary>
	''' <param name="sender"></param>
	''' <param name="e"></param>
	Public Shared Sub OnThreadException(ByVal sender As Object, ByVal e As Threading.ThreadExceptionEventArgs)

		CatchException(e.Exception)
		Application.Exit()

	End Sub

	''' <summary>
	''' 意図的にキャッチしない例外を全て処理する
	''' </summary>
	''' <param name="ex"></param>
	Public Shared Sub CatchException(ByVal ex As Exception)

		Dim message As String
		XmlSettings.LoadFromXmlFile()

		message = "システムエラーが発生しました" & vbNewLine & vbNewLine & "Exception=" & ex.GetType.Name & vbNewLine & "Message=" & ex.Message
		MessageBox.Show(message, XmlSettings.Instance.ApplicationName, MessageBoxButtons.OK, MessageBoxIcon.Error)

	End Sub

	Private Shared Sub Initialize()

		XmlSettings.LoadFromXmlFile()

		MyMutex = New System.Threading.Mutex(False, XmlSettings.Instance.ApplicationName)

		If MyMutex.WaitOne(0, False) = False Then

			MessageBox.Show("多重起動はできません", XmlSettings.Instance.ApplicationName, MessageBoxButtons.OK, MessageBoxIcon.Error)
			Exit Sub

		End If

		Dim context As AppContext = AppContext.GetInstance
		context.MainForm = New frmMain
		Application.Run(context)

	End Sub

End Class

File: PDFしおりリンクツール/PDF_Shiori/Module/clsWinApp.cs
using System;
using System.IO;
using System.Globalization;
using System.Windows.Forms;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PDF_Shiori
{
    public class AppContext : ApplicationContext
    {
		private static AppContext instance_ = new AppContext();

		private static void Main(string[] args)
		{
			
		}

		public static AppContext GetInstance()
		{
			return instance_;
		}

		/// <summary>
		/// 画面遷移時の動作
		/// </summary>
		/// <param name="mainForm"></param>
		/// <param name="toHide"></param>
		public void SetNextContext(System.Windows.Forms.Form mainForm, bool toHide = false)
		{
			System.Windows.Forms.Form tempForm = this.MainForm;
			Cursor orgCur = Cursor.Current;
			Cursor.Current = Cursors.WaitCursor;

			this.MainForm = mainForm;

			if (toHide == true)
			{
				try
				{
					if (tempForm != null)
					{
						tempForm.Hide();
					}
				}
				catch (Exception)
				{

					throw;
					//エラーメッセージを追記する
				}
			}
			else
			{
				try
				{
					tempForm.Close();
				}
				catch (Exception)
				{

					throw;
					//エラーメッセージを追記する
				}
			}

			this.MainForm.Show();
			tempForm = null;
			Cursor.Current = orgCur;
		}

    }

	/// <summary>
	/// メインルーチン
	/// </summary>
    public class clsWinApp
    {
		private static System.Threading.Mutex MyMutex;

		[STAThread]
		public static void Main()
		{
			//ビジュアルスタイルを有効にする
			Application.EnableVisualStyles();

			Initialize();
		}

		/// <summary>
		/// 例外ハンドラ
		/// </summary>
		/// <param name="sender"></param>
		/// <param name="e"></param>
		public static void OnThreadException(Object sender, System.Threading.ThreadExceptionEventArgs e)
		{
			CatchException(e.Exception);
			Application.Exit();
		}

		/// <summary>
		/// 意図的にキャッチしない例外を全て処理する
		/// </summary>
		/// <param name="ex"></param>
		public static void CatchException(Exception ex)
		{
			string message;
			XmlSettings.LoadFromXmlFile();

			message = "システムエラーが発生しました\n\n" + "Exception=" + ex.GetType().ToString() + "\nMessage=" + ex.Message;
			MessageBox.Show(message, XmlSettings.Instance.ApplicationName, MessageBoxButtons.OK, MessageBoxIcon.Error);
		}

		public static void Initialize()
		{
			XmlSettings.LoadFromXmlFile();

			MyMutex = new System.Threading.Mutex(false, XmlSettings.Instance.ApplicationName);

			if (MyMutex.WaitOne(0, false) == false)
			{
				MessageBox.Show("多重起動はできません", XmlSettings.Instance.ApplicationName, MessageBoxButtons.OK, MessageBoxIcon.Error);
				return;
			}

			AppContext context = AppContext.GetInstance();
			context.MainForm = new frmMenu();
			Application.Run(context);
		}
    }
}
